Jellyfin.Plugin.OpenSubtitles/OpenSubtitleDownloader.cs

```diff
@@ -207,124 +168,143 @@ await Task.Delay(250, cancellationToken)
         {
             if (string.IsNullOrWhiteSpace(id))
             {
-                throw new ArgumentNullException(nameof(id));
+                throw new ArgumentException("Missing param", nameof(id));
             }
 
+            if (_login?.User?.RemainingDownloads <= 0)
+            {
+                if (_limitReset < DateTime.UtcNow)
+                {
+                    _logger.LogDebug("Reset time passed, forcing a new login");
+                    // force login because the limit resets at midnight
+                    _login = null;
+                }
+                else
+                {
+                    throw new OpenApiException("OpenSubtitles download limit reached");
+                }
+            }
+
+            await Login(cancellationToken).ConfigureAwait(false);
+
             var idParts = id.Split('-', 3);
             var format = idParts[0];
             var language = idParts[1];
             var ossId = idParts[2];
 
-            var downloadsList = new[] { int.Parse(ossId, UsCulture) };
+            var fid = int.Parse(ossId, _usCulture);
 
-            await Login(cancellationToken).ConfigureAwait(false);
+            var info = await OpenSubtitlesHandler.OpenSubtitles.GetSubtitleLinkAsync(fid, _login, cancellationToken).ConfigureAwait(false);
 
-            if ((DateTime.UtcNow - _lastRateLimitException).TotalHours < 1)
+            if (!info.Ok)
             {
-                throw new RateLimitExceededException("OpenSubtitles rate limit reached");
-            }
+                switch (info.Code)
+                {
+                    case HttpStatusCode.NotAcceptable when info.Data.Remaining <= 0:
+                    {
+                        if (_login?.User != null)
+                        {
+                            _login.User.RemainingDownloads = 0;
+                        }
 
-            if (_rateLimitLeft == 0)
-            {
-                await Task.Delay(1000, cancellationToken)
-                    .ConfigureAwait(false);
-            }
+                        throw new OpenApiException("OpenSubtitles download limit reached");
+                    }
 
-            var downloadResponse = await OpenSubtitlesHandler.OpenSubtitles.DownloadSubtitlesAsync(downloadsList, cancellationToken).ConfigureAwait(false);
+                    case HttpStatusCode.Unauthorized:
+                        // JWT token expired, obtain a new one and try again?
```

Comment:
If it's JWT, shouldn't the expiration be in the token?

Reply:
It is, solved in edc7da120fcc23fc18bb7f7faaabc9eb1d112eaa